packages/components/src/components/domainproperties/issues/IssuesListDefDesignerPanels.tsx

```diff
@@ -0,0 +1,192 @@
+    successBsStyle?: string;
+    saveBtnText?: string;
+}
+
+interface State {
+    model: IssuesListDefModel;
+}
+
+class IssuesDesignerPanelsImpl extends React.PureComponent<Props & InjectedBaseDomainDesignerProps, State> {
+    constructor(props: Props & InjectedBaseDomainDesignerProps) {
+        super(props);
+
+        this.state = {
+            model: props.initModel || IssuesListDefModel.create({}),
+        };
+    }
+
+    onPropertiesChange = (model: IssuesListDefModel) => {
+        const { onChange } = this.props;
+
+        this.setState(
+            produce((draft: Draft<State>) => {
+                draft.model = model;
+            }),
+            () => {
+                if (onChange) {
+                    onChange(model);
+                }
+            }
+        );
+    };
+
+    onDomainChange = (domain: DomainDesign, dirty: boolean) => {
+        const { onChange } = this.props;
+
+        this.setState(
+            produce((draft: Draft<State>) => {
+                draft.model.domain = domain;
+            }),
+            () => {
+                if (onChange && dirty) {
+                    onChange(this.state.model);
+                }
+            }
+        );
+    };
+
+    onFinish = () => {
+        const { setSubmitting } = this.props;
+        const { model } = this.state;
+        const isValid = IssuesListDefModel.isValid(model);
+
+        this.props.onFinish(isValid, this.saveDomain);
+    };
+
+    saveDomain = () => {
+        const { setSubmitting } = this.props;
+        const { model } = this.state;
+
+        saveDomain(model.domain, model.domainKindName, model.getOptions(), model.name)
+            .then(response => {
+
+                this.setState(
+                    produce((draft: Draft<State>) => {
+                        const updatedModel = draft.model;
+                        updatedModel.exception = undefined;
+                        updatedModel.domain = response;
+                    }),
+                    () => {
+                        setSubmitting(false, () => {
+                            const { model } = this.state;
+                            this.props.onComplete(model);
+                    })
+                });
+            })
+            .catch(response => {
+                const exception = resolveErrorMessage(response);
+
+                this.setState(
+                    produce((draft: Draft<State>) => {
+                        const updatedModel = draft.model;
+                        if (exception) {
+                            updatedModel.exception = exception;
+                        } else {
+                            updatedModel.exception = undefined;
+                            updatedModel.domain = response;
+                        }
+                    }),
+                    () => {
+                        setSubmitting(false, () => {
+                            this.props.onComplete(model);
```

Comment:
They probably won't be different, but instead of using `model` here, we should probably use `this.state.model` since this is in the callback that changes that model.

Reply:
Actually, I don't think we want to be calling the `this.props.onComplete` here in the error / catch block. I think this setState callback can be removed.

Reply:
Yeah i had not included that initially, but then added it back for some reason.
now removed!